collect_app/src/main/java/org/odk/collect/android/application/initialization/ApplicationInitializer.java

```diff
@@ -0,0 +1,107 @@
+package org.odk.collect.android.application.initialization;
+
+import android.app.Application;
+import android.content.SharedPreferences;
+import android.util.Log;
+
+import androidx.appcompat.app.AppCompatDelegate;
+
+import com.crashlytics.android.Crashlytics;
+import com.evernote.android.job.JobManager;
+import com.evernote.android.job.JobManagerCreateException;
+
+import net.danlew.android.joda.JodaTimeAndroid;
+
+import org.odk.collect.android.BuildConfig;
+import org.odk.collect.android.application.Collect;
+import org.odk.collect.android.jobs.CollectJobCreator;
+import org.odk.collect.android.preferences.AdminSharedPreferences;
+import org.odk.collect.android.preferences.AutoSendPreferenceMigrator;
+import org.odk.collect.android.preferences.GeneralSharedPreferences;
+import org.odk.collect.android.utilities.LocaleHelper;
+import org.odk.collect.android.utilities.NotificationUtils;
+
+import java.util.Locale;
+
+import timber.log.Timber;
+
+public class ApplicationInitializer {
+
+    private final Application context;
+    private final CollectJobCreator collectJobCreator;
+    private final SharedPreferences metaSharedPreferences;
+    private final GeneralSharedPreferences generalSharedPreferences;
+    private final AdminSharedPreferences adminSharedPreferences;
+
+    public ApplicationInitializer(Application context, CollectJobCreator collectJobCreator, SharedPreferences metaSharedPreferences) {
+        this.context = context;
+        this.collectJobCreator = collectJobCreator;
+        this.metaSharedPreferences = metaSharedPreferences;
+
+        generalSharedPreferences = GeneralSharedPreferences.getInstance();
+        adminSharedPreferences = AdminSharedPreferences.getInstance();
+    }
+
+    public void initializePreferences() {
+        performMigrations();
+        reloadSharedPreferences();
+    }
+
+    public void initializeFrameworks() {
```

Comment:
I think it would be good to factor out more methods here like `initializeTimber`/`initializeJobManager` etc. It might be strange to create separate methods for those one line initializations so it's ok to keep them in one method.

Reply:
I think you're right it's pretty hard to read that method!